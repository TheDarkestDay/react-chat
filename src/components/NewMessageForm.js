import React, { Component } from 'react';
import Button from 'material-ui/Button';
import Input from 'material-ui/Input';
import Paper from 'material-ui/Paper';
import { withStyles } from 'material-ui/styles';

const styles = () => ({
  newMessageFieldWrapper: {
    padding: '16px 24px'
  }
});

class NewMessageForm extends Component {
  render() {
    const { classes, isMember, joinChat } = this.props;

    return (
      <Paper className={classes.newMessageFieldWrapper}>
        {isMember &&
          (<form>
            <Input placeholder="Type your message..." fullWidth />
          </form>)
        }
        {!isMember &&
          (<Button onClick={joinChat} variant="raised" color="primary" fullWidth> Join </Button>)
        }
      </Paper>
    );
  }
}

export default withStyles(styles)(NewMessageForm);