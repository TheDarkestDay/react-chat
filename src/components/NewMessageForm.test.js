/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NewMessageForm from './NewMessageForm';

it('renders without crashing', () => {
  const newMessageFormPropsMock = {
    disabled: false,
    isAllowedToSendMessages: true,
    joinChat: () => {},
    sendMessage: () => {},
  };

  const div = document.createElement('div');
  ReactDOM.render(<NewMessageForm {...newMessageFormPropsMock} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should match snapshot', () => {
  const newMessageFormPropsMock = {
    disabled: false,
    isAllowedToSendMessages: true,
    joinChat: () => {},
    sendMessage: () => {},
  };

  const tree = renderer.create(<NewMessageForm {...newMessageFormPropsMock} />).toJSON();
  expect(tree).toMatchSnapshot();
});
