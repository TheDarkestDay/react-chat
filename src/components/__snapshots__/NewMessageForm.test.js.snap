// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="MuiPaper-root-45 MuiPaper-shadow2-49 MuiPaper-rounded-46 NewMessageForm-newMessageFieldWrapper-44"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiInput-root-72 MuiInput-fullWidth-80 MuiInput-inkbar-74 MuiInput-underline-78"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      <input
        aria-invalid={undefined}
        aria-required={undefined}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-81"
        defaultValue={undefined}
        disabled={false}
        id={undefined}
        name={undefined}
        onChange={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Type your message..."
        readOnly={undefined}
        required={undefined}
        rows={undefined}
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
